types/node.lua

```diff
@@ -686,16 +907,99 @@ function nodetypes.vbox:append (box)
    self.depth = lastdepth
 end
 
-nodetypes.migrating = pl.class(nodetypes.hbox)
-nodetypes.migrating.type = "migrating"
-nodetypes.migrating.material = {}
-nodetypes.migrating.value = {}
-nodetypes.migrating.nodes = {}
-
-function nodetypes.migrating:__tostring ()
+--- A migrating node is a node that can be moved from one frame to another.
+-- Typically, footnotes are migrating nodes.
+--
+-- Derived from `hbox`.
+--
+-- Properties ìs_migrating, is_hbox and is_box are true.
+--
+-- @type migrating
+-- @usage
+-- SILE.types.node.migrating({ material =  ... })
```

Comment:
This class can be passed anything (no constructor to enforce anything), and is thus _very_ loosely defined.

What material is supposed to be is also unclear (a list of vboxes? insertions-specific nodes)?